### PR TITLE
Add guide for builtin libraries

### PR DESCRIPTION
**Pull request by msujew**
Closes https://github.com/langium/langium-website/issues/68

### CODE REVIEW

File: hugo/content/guides/builtin-library.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+---
+title: "Builtin Libraries"
+weight: 300
+---
+
+Languages usually offer their users some high-level programming features that they do not have to define themselves.
+For example, TypeScript provides users with typings for globally accessible variables such as the `window`, `process` or `console` objects.
+They are part of the JavaScript runtime, and not defined by any user or a package they might import.
+Instead, these features are contributed through what we call a builtin library.
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:9]:
```suggestion
Instead, these features are contributed through what we call builtin libraries.
```

File: hugo/content/guides/builtin-library.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+Fortunately, loading a builtin library in Langium is very simple. We first start off with defining the source code of the library using the *hello world* language from the [getting started guide](/docs/getting-started):
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:11]:
```suggestion
Loading a builtin library in Langium is very simple. We first start off with defining the source code of the library using the *hello world* language from the [getting started guide](/docs/getting-started):
```
Can drop the use of fortunately in this case.

File: hugo/content/guides/builtin-library.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+```ts
+export const builtinHelloWorld = `
+person Jane
+person John
+`.trimLeft();
+```
+
+Next, we load our builtin library code through the `loadAdditionalDocuments` provided by the `DefaultWorkspaceManager`:
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:20]:
```suggestion
Next, we load our builtin library code through the `loadAdditionalDocuments` method provided by the `DefaultWorkspaceManager`:
```

File: hugo/content/guides/builtin-library.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+```ts
+import {
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:23]: Might be nice to clarify where this content should be located.

File: hugo/content/guides/builtin-library.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        // Load our library using the `builtin` URI schema
+        collector(this.documentFactory.fromString(builtinHelloWorld, URI.parse('builtin:///library.hello')));
+    }
+}
+```
+
+As a last step, we have to bind our newly created workspace manager:
+
+```ts
+export const HelloWorldSharedModule: Module<HelloWorldSharedServices, DeepPartial<LangiumSharedServices>> = {
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:52]: Same here, also nice to indicate where a good place to put this is.

File: hugo/content/guides/builtin-library.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    workspace: {
+        WorkspaceManager: (services) => new HelloWorldWorkspaceManager(services)
+    }
+}
+```
+
+Be aware that this shared module is not injected by default. You have to add it manually to the `inject` call for the shared injection container.
```
Comment (montymxb) [inline hugo/content/guides/builtin-library.md:59]: Perhaps add a few lines to show how to add this as well?

**Review by montymxb — COMMENTED** [review]
Some general feedback. I didn't know that you could provide a FileSystemProvider to allow inspecting of in-memory contents, very neat 😎 .

File: hugo/content/guides/builtin-library.md (hunk 4 above)
Comment (msujew) [reply hugo/content/guides/builtin-library.md:23]: The workspace manager is fairly isolated, so users can place it anywhere they like. I wouldn't impose a folder/directory structure for them in this simple guide.

**Review by spoenemann — APPROVED** [review]
Let's ship this so it's ready for the v1.0 release.

**Pull request merged**